### ISSUE TITLE
Busybox Check Broken After #39 

### ISSUE THREAD

**Issue by hopez13** [Magisk-Modules-Alt-Repo/chroot-distro#42]
Apatch Comes With Prebuilt  Busybox So No Extra Module Is required 
But  #39  Broke The Detection and Says Busybox Not Found

Reverting Back To 1.1.4 Fixes The Issue

**Comment by defectly** [comment]
that sucks..

**Comment by jjkola** [comment]
Thanks for the report. Unfortunately, there are too many variations with the enabled features of Busybox between different providers. There are already Busybox ndk, KernelSU, and now APatch. Also, each of them may have different version of Busybox.

And that's not yet counting Termux, ADB, and other commandline providers, which each provide their own set environments and differences.

Thus, to ensure at least somewhat consistent running environment between different environments, `chroot-distro` requires Busybox ndk installed.

**Comment by jjkola** [comment]
Also, to be clear: `chroot-distro` has required Busybox ndk from first release. It just has happened to work with other Busybox providers before the latest versions.

But, there have already been several reports because of the differences, with your report being the latest. And even some reports where there wasn't any Busybox installed.

Not to be rude, but at least I'm doing this in my free time. Thus, I want to keep the amount of work manageable. By providing a consistent environment, I will be able to contribute fixes and new features whenever I have time. If I tried to support every different environment, then all my time would go trying to work with differences, and I wouldn't even be able to test all the different environments, thus creating more work for me, and worse experience for other users.

**Comment by jjkola** [comment]
Also, to be clear: `chroot-distro` has required Busybox ndk from first release. It just has happened to work with other Busybox providers before the latest versions.

But, there have already been several reports because of the differences, with your report being the latest. And even some reports where there wasn't any Busybox installed.

Not to be rude, but at least I'm doing this in my free time. Thus, I want to keep the amount of work manageable. By providing a consistent environment, I will be able to contribute fixes and new features whenever I have time. If I tried to support every different environment, then all my time would go trying to work with differences, and I wouldn't even be able to test all the different environments, thus creating more work for me, and worse experience for other users.

If you still feel that your environment (APatch without Busybox ndk) should be supported, then I would like to hear your thoughts on how we support your environment while keeping the existing feature set and ensuring your environment keeps working with future versions? Or, you can also close the issue if you think I have provided enough reasons to use Busybox ndk. Or, if you think that `chroot-distro` is not something you want to use, you are free to choose something else. For example, I myself tried several options before settling with this. But if you choose the last one, I hope that you will still close the issue so the issue will not be left hanging.

**Comment by jjkola** [comment]
Thank you for your understanding

**Comment by defectly** [comment]
> _For some reason the Github client copied text from the previous reply here, it has been removed..._
> 
> If you still feel that your environment (APatch without Busybox ndk) should be supported, then I would like to hear your thoughts on how we support your environment while keeping the existing feature set and ensuring your environment keeps working with future versions? Or, you can also close the issue if you think I have provided enough reasons to use Busybox ndk. Or, if you think that `chroot-distro` is not something you want to use, you are free to choose something else. For example, I myself tried several options before settling with this. But if you choose the last one, I hope that you will still close the issue so the issue will not be left hanging.

maybe you could add some questions in installation process like which root template to use or path to busybox..

**Comment by jjkola** [comment]
If it was that easy then yes, but unfortunately there is quite much variation even in such a basic thing as parsing the command line options. If KernelSU and APatch started to provide at least the same set of the features as Busybox ndk, then we can start talking. But, that is a problem which needs to be addressed in the respective projects, and can't be resolved in this project. Or, should I say it can be addressed in this project but then this project would have to change from shell script to a binary package.

**Comment by jjkola** [comment]
The thing is that they may have the same binary but clearly they have forgotten to add all the same links as is provided by Busybox NDK. This should be fixed at their respective respective projects, but since you guys insist, I will "fix" it. I could have used this time to fix other issues (broken links etc.).

Also, since you insist supporting KernelSU and APatch without Busybox NDK then I will add nomenclature that you guys are running in environment where it may experience degraded experience, as I can't directly support them (ie. you will have to report problems, and I will have to spend time fixing the problems).

**Comment by jjkola** [comment]
Can you guys try a [fix](https://github.com/jjkola/chroot-distro/tree/fix/kernelsu_and_apatch_without_ndk)? If it works then I will push it forward, and create a new version after #35 is fixed.

**Comment by jjkola** [comment]
> I will test it
> 
> but here [jjkola@3020822](https://github.com/jjkola/chroot-distro/commit/3020822011ea78bc77fae2cfebe26d38ee226912)
> 
> I think changing it at every place is not necessary you can just add one line patch I sent before
> 
> `getopt() { busybox getopt "$@"; }`

I tested this, and it seems to work. I have now changed the code to use this one.

**Comment by osm0sis** [comment]
busybox isn't normally even exposed on the $PATH from the root solutions without a separate module. 

If you want something more robust to cover all of that, check out what I did for wget here: 
https://github.com/osm0sis/PlayIntegrityFork/blob/main/module/autopif.sh#L25-L45

**Comment by jjkola** [comment]
I think I nailed the implementation yesterday night. I will have to test it against ADB shell to ensure things should work as I expect them. And on the positive side, I fixed the XZ compression problem for real this time.

**Comment by jjkola** [comment]
Hopefully the new PR will solve all current compatibility problems, and make the future changes easier to debug. Could have gone the easy route but wanted to make solution which will last so that I will not have to touch that part of the code unless some completely new functionality comes which requires more compatibility additions.
